File: Source/Kernel/Contracts/ReadModels/ReadModelSnapshotGrouping.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Cratis.Chronicle.Contracts.ReadModels;

/// <summary>
/// Represents how the events that shaped a read model are grouped into snapshots.
/// </summary>
/// <remarks>
/// Correlation is zero so that a request that leaves the field unset groups by correlation, which is
/// what every caller got before there was a choice.
/// </remarks>
public enum ReadModelSnapshotGrouping
{
    /// <summary>
    /// One snapshot per correlation - the events that were applied as a single action.
    /// </summary>
    Correlation = 0,

    /// <summary>
    /// One snapshot per event, so every snapshot moves the read model by exactly one thing that happened.
    /// </summary>
    Event = 1
}

File: Source/Workbench/Components/TimeScrubberDialog/ReadModelSnapshotGrouping.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

/**
 * How the events that shaped a read model are grouped into snapshots.
 *
 * The values name the kernel's own grouping, which the query takes as a string - an enum query
 * parameter is dropped by the proxy generator, so the contract across the wire is the name.
 */
export enum ReadModelSnapshotGrouping {
    /** One snapshot per correlation - the events that were applied as a single action. */
    Correlation = 'Correlation',

    /** One snapshot per event, so every snapshot moves the read model by exactly one thing that happened. */
    Event = 'Event'
}
